pkg/controller/replication_controller.go

```diff
@@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
 		return nil
 	}
+	if len(controllers) > 1 {
```

Comment:
In this scenario though there doesn't seem to be a benefit to not being deterministic, since the fix will be to disable the newer one or the older one (probably the newer one).

> On Jun 26, 2015, at 7:05 PM, Prashanth B <notifications@github.com> wrote:
> 
> In pkg/controller/replication_controller.go:
> 
> > @@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
> >  		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
> >  		return nil
> >  	}
> > +	if len(controllers) > 1 {
> I feel like you don't want to rely on behavior in the rc manager to give you deterministic outcomes when rcs shouldn't overlap. I'm not really against sorting, just the people might start relying on it and when we do this right, it'll break.
> 
> —
> Reply to this email directly or view it on GitHub.
>